libpysal/weights/util.py

```diff
@@ -1117,8 +1118,12 @@ def get_points_array_from_shapefile(shapefile):
            [ 8.33265837, 14.03162401],
            [ 9.01226541, 13.81971908]])
     """
-
-    f = psopen(shapefile)
+    if isinstance(shapefile, str):
+        f = psopen(shapefile)
+    elif isinstance(shapefile, gpd.GeoDataFrame):
+        f = shapefile['geometry']
```

Comment:
```suggestion
        f = shapefile.geometry
```

GeoDataFrame can have an active geometry column named differently than `'geometry'`, we always need to access it via an attribute.